src/Integration/Binding/DotNetBindingConfigProvider.cs

```diff
@@ -44,83 +45,98 @@ internal class DotNetBindingConfigProvider : IBindingConfigProvider
             this.projectName = projectName;
             this.logger = logger;
+
+            this.ruleSetGenerator = ruleSetGenerator;
+            this.nuGetPackageInfoGenerator = nuGetPackageInfoGenerator;
         }
 
         public bool IsLanguageSupported(Language language)
         {
             return Language.CSharp.Equals(language) || Language.VBNET.Equals(language);
         }
 
-        public async Task<IBindingConfigFile> GetConfigurationAsync(SonarQubeQualityProfile qualityProfile, string organizationKey, Language language, CancellationToken cancellationToken)
+        public Task<IBindingConfigFile> GetConfigurationAsync(SonarQubeQualityProfile qualityProfile, string UNUSED_organizationKey_TODO_REMOVE, Language language, CancellationToken cancellationToken)
         {
-            if(!IsLanguageSupported(language))
+            if (!IsLanguageSupported(language))
             {
                 throw new ArgumentOutOfRangeException(nameof(language));
             }
 
+            return DoGetConfigurationAsync(qualityProfile, language, cancellationToken);
+        }
+
+        private async Task<IBindingConfigFile> DoGetConfigurationAsync(SonarQubeQualityProfile qualityProfile, Language language, CancellationToken cancellationToken)
+        {
             var serverLanguage = language.ServerLanguage;
             Debug.Assert(serverLanguage != null,
                 $"Server language should not be null for supported language: {language.Id}");
 
-            // Generate the rules configuration file for the language
-            var roslynProfileExporter = await WebServiceHelper.SafeServiceCallAsync(() =>
-                this.sonarQubeService.GetRoslynExportProfileAsync(qualityProfile.Name,
-                    organizationKey, serverLanguage, cancellationToken),
-                this.logger);
-            if (roslynProfileExporter == null)
-            {
-                this.logger.WriteLine(string.Format(Strings.SubTextPaddingFormat,
-                    string.Format(Strings.QualityProfileDownloadFailedMessageFormat, qualityProfile.Name,
-                        qualityProfile.Key, language.Name)));
-                return null;
-            }
+            // First, fetch the active rules
+            var activeRules = await WebServiceHelper.SafeServiceCallAsync(
+                () => sonarQubeService.GetRulesAsync(true, qualityProfile.Key, cancellationToken), logger);
 
-            var tempRuleSetFilePath = Path.GetTempFileName();
-            File.WriteAllText(tempRuleSetFilePath, roslynProfileExporter.Configuration.RuleSet.OuterXml);
-            RuleSet ruleSet = RuleSet.LoadFromFile(tempRuleSetFilePath);
-
-            // Give up if the quality profile is empty
-            if (ruleSet == null ||
-                ruleSet.Rules.Count == 0 ||
-                ruleSet.Rules.All(rule => rule.Action == RuleAction.None))
+            // Give up if the quality profile is empty - not point in fetching anything else
```

Comment:
This function is already getting quite long, maybe extract the 3 checks to 3 separate functions? (each with the logging in it)

Reply:
I've extracted a couple of methods.